### PR TITLE
mercurial: 4.9.1 -> 5.2.1 + python3  🚀

### PR DESCRIPTION
**Pull request by Mic92**
###### Motivation for this change

Get python2 out of my nix profile

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/applications/version-management/tortoisehg/default.nix [hunk 1]
```diff
@@ -1,51 +1,46 @@
...
-        mercurialSrc = fetchurl rec {
-          meta.name = "mercurial-${meta.version}";
-          meta.version = tortoisehgSrc.meta.version;
-          url = "https://mercurial-scm.org/release/${meta.name}.tar.gz";
-          sha256 = "1y60hfc8gh4ha9sw650qs7hndqmvbn0qxpmqwpn4q18z5xwm1f19";
-        };
-      };
-
-in python2Packages.buildPythonApplication {
+  tortoiseMercurial = mercurial.overrideDerivation (old: {
```
Comment (FRidh) [inline pkgs/applications/version-management/tortoisehg/default.nix]: OverridePythonAttrs

File: pkgs/applications/version-management/tortoisehg/default.nix [hunk 2]
```diff
@@ -1,51 +1,46 @@
...
-    propagatedBuildInputs = with python2Packages; [ qscintilla iniparse ];
+    propagatedBuildInputs = with python3Packages; [ tortoiseMercurial qscintilla-qt5 iniparse ];
+    nativeBuildInputs = [ qt5.wrapQtAppsHook ];
 
     doCheck = false; # tests fail with "thg: cannot connect to X server"
     dontStrip = true;
     buildPhase = "";
     installPhase = ''
-      ${python2Packages.python.executable} setup.py install --prefix=$out
+      ${python3Packages.python.executable} setup.py install --prefix=$out
```
Comment (FRidh) [inline pkgs/applications/version-management/tortoisehg/default.nix]: Is it still not possible to build a wheel?

**Comment by Mic92** [comment]
fixed

**Comment by danbst** [comment]
I don't object this merge. There are problems with `tortoisehg` which may be fixed in further PRs:
- tortoisehg is now slower. Startup, repo opening, any operation. I don't quite understand why. Is it because of Qt5?
- `iniparse` package isn't found at runtime. To reproduce, open File->Settings. It is not critical though

**Pull request merged**